pkg/store/proxy_merge.go

```diff
@@ -166,12 +167,21 @@ func (d *responseDeduplicator) At() *storepb.SeriesResponse {
 	return d.bufferedResp[d.buffRespI]
 }
 
+type seriesResponseWithStoreLabelset struct {
+	labelset string
+	response *storepb.SeriesResponse
+}
+
+type proxyResponseLoserTree struct {
+	tree *losertree.Tree[seriesResponseWithStoreLabelset, respSet]
+}
+
 // NewProxyResponseLoserTree returns heap that k-way merge series together.
-// It's agnostic to duplicates and overlaps, it forwards all duplicated series in random order.
-func NewProxyResponseLoserTree(seriesSets ...respSet) *losertree.Tree[*storepb.SeriesResponse, respSet] {
-	var maxVal *storepb.SeriesResponse = storepb.NewSeriesResponse(nil)
+// It's agnostic to duplicates and overlaps, it forwards all duplicated series ordered by the labelset of their endpoint.
+func NewProxyResponseLoserTree(seriesSets ...respSet) *proxyResponseLoserTree {
+	var maxVal seriesResponseWithStoreLabelset = seriesResponseWithStoreLabelset{}
```

Comment:
```suggestion
	var maxVal = seriesResponseWithStoreLabelset{}
```

Type can be inferred from the right-side